### PR TITLE
Bump golang 1.18 and k8s 1.24

### PR DESCRIPTION
**Pull request by SchSeba**
Signed-off-by: Sebastian Sch <sebassch@gmail.com>

### CODE REVIEW

**Review by adrianchiris — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/k8snetworkplumbingwg/sriov-network-device-plugin
 
-go 1.13
+go 1.17
```
Comment (e0ne) [inline go.mod:3]: Why thre is 1.17 instead of 1.18?

File: .github/workflows/build-test-lint.yml [hunk 2]
```diff
@@ -5,13 +5,13 @@ jobs:
     name: build
     strategy:
       matrix:
-        go-version: [1.13.x, 1.14.x, 1.15.x, 1.16.x]
+        go-version: [1.17.x, 1.18.x]
```
Comment (e0ne) [inline .github/workflows/build-test-lint.yml:8]: Do we still need to support 1.17.x?

File: go.mod (hunk 1 above)
Comment (SchSeba) [reply go.mod:3]: there is no 1.18 version in go.mod files yet

File: .github/workflows/build-test-lint.yml (hunk 2 above)
Comment (SchSeba) [reply .github/workflows/build-test-lint.yml:8]: I think it's in the build verification (we only publish 1.18 built container images)

**Pull request merged**